### PR TITLE
added initial dispenser subsystem

### PR DESCRIPTION
**Pull request by yutytuty**

### CODE REVIEW

File: src/main/java/frc/robot/Subsystems/Dispenser.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    private static Dispenser instance;
+
+    private Dispenser(WPI_TalonSRX talon) {
+        super("dispenser", talon);
+        this.talon = talon;
+        this.encoder = new Encoder(RobotMap.DIO.DISPENSER_ENCODER_POS, RobotMap.DIO.DISPENSER_ENCODER_NEG);
+    }
+
+    public static Dispenser getInstance() {
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Subsystems/Dispenser.java:21]: not the right place for the `getInstance()` function

File: src/main/java/frc/robot/Subsystems/Dispenser.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package frc.robot.Subsystems;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.spikes2212.command.genericsubsystem.MotoredGenericSubsystem;
+import edu.wpi.first.wpilibj.Encoder;
+import frc.robot.RobotMap;
+
+public class Dispenser extends MotoredGenericSubsystem {
+
+    private WPI_TalonSRX talon;
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Subsystems/Dispenser.java:10]: why are you saving this vatiable?

File: src/main/java/frc/robot/Subsystems/Dispenser.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        this.talon = talon;
+        this.encoder = new Encoder(RobotMap.DIO.DISPENSER_ENCODER_POS, RobotMap.DIO.DISPENSER_ENCODER_NEG);
+    }
+
+    public static Dispenser getInstance() {
+        if (instance == null) {
+            instance = new Dispenser(new WPI_TalonSRX(RobotMap.CAN.DISPENSER_TALON));
+        }
+        return instance;
+    }
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Subsystems/Dispenser.java:26]: add configureDashboard()

**Review by tal-sitton — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Subsystems/Dispenser.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package frc.robot.Subsystems;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.spikes2212.command.genericsubsystem.MotoredGenericSubsystem;
+import edu.wpi.first.wpilibj.Encoder;
+import frc.robot.RobotMap;
+
+public class Dispenser extends MotoredGenericSubsystem {
+
+    private static final double DISTANCE_PER_PULSE = 20.32 / 4096.0;
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Subsystems/Dispenser.java:10]: WHY?

File: src/main/java/frc/robot/Subsystems/Dispenser.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    private Dispenser(WPI_TalonSRX talon) {
+        super("dispenser", talon);
+        this.encoder = new Encoder(RobotMap.DIO.DISPENSER_ENCODER_POS, RobotMap.DIO.DISPENSER_ENCODER_NEG);
+        this.encoder.setDistancePerPulse(DISTANCE_PER_PULSE);
+    }
+
+    @Override
+    public void configureDashboard() {
+        rootNamespace.putNumber("encoder ticks", encoder::get);
+        rootNamespace.putNumber("encoder distance", encoder::getDistance);
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Subsystems/Dispenser.java:32]: WHY?

**Review by tal-sitton — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Subsystems/Dispenser.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
+package frc.robot.Subsystems;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.spikes2212.command.genericsubsystem.MotoredGenericSubsystem;
+import edu.wpi.first.wpilibj.Encoder;
+import frc.robot.RobotMap;
+
+public class Dispenser extends MotoredGenericSubsystem {
+
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Subsystems/Dispenser.java:9]: add Limelight

**Review by tal-sitton — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Subsystems/Dispenser.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
+package frc.robot.Subsystems;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.spikes2212.command.genericsubsystem.MotoredGenericSubsystem;
+import com.spikes2212.util.Limelight;
+import edu.wpi.first.wpilibj.Encoder;
+import frc.robot.RobotMap;
+
+public class Dispenser extends MotoredGenericSubsystem {
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Subsystems/Dispenser.java:9]: add docs

**Review by tal-sitton — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Subsystems/Dispenser.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
+package frc.robot.Subsystems;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.spikes2212.command.genericsubsystem.MotoredGenericSubsystem;
+import com.spikes2212.util.Limelight;
+import edu.wpi.first.wpilibj.Encoder;
+import frc.robot.RobotMap;
+
+/**
+ * drops cargo pieces fed to the dispenser from the transfer system into the lower hub.
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Subsystems/Dispenser.java:10]: rewrite

**Review by tal-sitton — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Subsystems/Dispenser.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
+package frc.robot.Subsystems;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.spikes2212.command.genericsubsystem.MotoredGenericSubsystem;
+import com.spikes2212.util.Limelight;
+import edu.wpi.first.wpilibj.Encoder;
+import frc.robot.RobotMap;
+
+/**
+ * drops cargo into the lower hub
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Subsystems/Dispenser.java:10]: that describes the command, not the subsystem

**Review by tal-sitton — CHANGES_REQUESTED** [review]

**Comment by yutytuty** [comment]
removed subsystem. Deleting.

**Pull request closed without merging**